pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go

```diff
@@ -164,8 +167,61 @@ func (r *HTTPRouteReconciler) shouldHandleParentRef(
 	return class.Spec.ControllerName == controllerName, nil
 }
 
+// routesForGateway returns a function that calculates which routes might
+// be affected by changes in a Gateway so they can be reconciled.
+func routesForGateway(l logr.Logger, client kube_client.Client) kube_handler.MapFunc {
+	l = l.WithName("routesForGateway")
+
+	return func(obj kube_client.Object) []kube_reconcile.Request {
+		gateway, ok := obj.(*gatewayapi.Gateway)
+		if !ok {
+			l.Error(nil, "unexpected error converting to be mapped %T object to Gateway", obj)
+			return nil
+		}
+
+		var routes gatewayapi.HTTPRouteList
+		if err := client.List(context.Background(), &routes); err != nil {
+			l.Error(err, "unexpected error listing HTTPRoutes in cluster")
+			return nil
+		}
+
+		var requests []kube_reconcile.Request
+		for _, route := range routes.Items {
+			for _, parentRef := range route.Spec.ParentRefs {
+				// We're looking at all HTTPRoutes, at some point one may
+				// reference a different kind of object.
+				matches := *parentRef.Group == gatewayapi.GroupName &&
+					*parentRef.Kind == "Gateway"
+
+				referencedNamespace := route.Namespace
+				if parentRef.Namespace != nil {
+					referencedNamespace = string(*parentRef.Namespace)
+				}
+
+				matches = matches &&
+					(referencedNamespace == gateway.Namespace) &&
+					(string(parentRef.Name) == gateway.Name)
```

Comment:
Would this code be more readable with a bunch of `if ... { continue }` instead of building a more and more complex condition?

Reply:
It's unreadable anyone you write it inline IMO :shrug: I'm going to put the check in a separate function.